pkg/flow/grpc-internal.go

```diff
@@ -90,19 +91,45 @@ func (internal *internal) ReportActionResults(ctx context.Context, req *grpc.Rep
 func (internal *internal) ActionLog(ctx context.Context, req *grpc.ActionLogRequest) (*emptypb.Empty, error) {
 	internal.sugar.Debugf("Handling gRPC request: %s", this())
 
-	t := time.Now()
-
 	cached, err := internal.getInstance(ctx, req.GetInstanceId())
 	if err != nil {
 		internal.sugar.Error(err)
 		return nil, err
 	}
 
+	rt, err := internal.edb.InstanceRuntime(ctx, cached.Instance.Runtime)
```

Comment:
This is probably fine for now. But we should make a note because this will hurt performance slightly.